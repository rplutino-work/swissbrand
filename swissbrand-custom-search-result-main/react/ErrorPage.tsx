import React from 'react'
import { useRuntime } from 'vtex.render-runtime'

const ErrorPage: StorefrontFunctionComponent = () => {
    const runtime = useRuntime()
    const term = runtime?.query?._q

  

        if(term){
          return (
            <>
              <div className={`vtex-flex-layout-0-x-ProductErrorPage--wrapper`}>
              <h1 className={`vtex-store-components-3-x-ProductErrorPage--title`}>Su búsqueda de "{term}" no arrojó ningún resultado.</h1>
              <p className={`vtex-store-components-3-x-ProductErrorPage--subtitle`}>Compruebe los términos introducidos. Intenta utilizar una sola palabra. Utilice términos genéricos en la búsqueda. Busque utilizar sinónimos al término deseado.</p>
              <p className={`vtex-store-components-3-x-ProductErrorPage--text`}>Tambien podés visiar alguno de los productos recomendados</p>
            </div>
            </>
          )
        }

        return (
          <>
            <div className={`vtex-flex-layout-0-x-PageErrorPage--wrapper`}>
            <h1 className={`vtex-store-components-3-x-PageErrorPage--title`}>¡Perdón!, no sabemos lo que pasó.</h1>
            <p className={`vtex-store-components-3-x-PageErrorPage--text`}>La página que estás buscando no la encontramos. ¡Pero no te vayas! Te recomendamos algunos productos</p>
          </div>
          </>
        ) 
}

export default ErrorPage
