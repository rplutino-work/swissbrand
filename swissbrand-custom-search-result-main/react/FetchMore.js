import React from 'react'
import { path } from 'ramda'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import FetchMoreButton from './components/loaders/FetchMoreButton'
import LoadingSpinner from './components/loaders/LoadingSpinner'
import { PAGINATION_TYPE } from './constants/paginationType'
import { useFetchMore } from './hooks/useFetchMore'

import { useSearchPage } from 'vtex.search-page-context/SearchPageContext'

import styles from './searchResult.css'

const FetchMore = ({ htmlElementForButton = 'button '}) => {
  const { pagination, searchQuery, maxItemsPerPage, page } = useSearchPage()
  const products = path(['data', 'productSearch', 'products'], searchQuery)
  const recordsFiltered = path(
    ['data', 'productSearch', 'recordsFiltered'],
    searchQuery
  )
  const fetchMore = path(['fetchMore'], searchQuery)
  const queryData = {
    query: path(['variables', 'query'], searchQuery),
    map: path(['variables', 'map'], searchQuery),
    orderBy: path(['variables', 'orderBy'], searchQuery),
    priceRange: path(['variables', 'priceRange'], searchQuery),
  }

  const { handleFetchMoreNext, loading, to } = useFetchMore({
    page,
    recordsFiltered,
    maxItemsPerPage,
    fetchMore,
    products,
    queryData,
  })

  const isShowMore = pagination === PAGINATION_TYPE.SHOW_MORE

  if (isShowMore) {
    return (
      <div
        className={classNames(
          styles['buttonShowMore--layout'],
          'w-100 flex justify-center'
        )}
      >
        <FetchMoreButton
          products={products}
          to={to}
          recordsFiltered={recordsFiltered}
          onFetchMore={handleFetchMoreNext}
          loading={loading}
          showProductsCount={false}
          htmlElementForButton={htmlElementForButton}
        />
      </div>
    )
  }

  return <LoadingSpinner loading={loading} />
}

FetchMore.propTypes = {
  /* html element to render for fetch more button */
  htmlElementForButton: PropTypes.string,
}

export default FetchMore
