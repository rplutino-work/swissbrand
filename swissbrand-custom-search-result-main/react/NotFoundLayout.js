import React, { useMemo } from 'react'
import { SearchPageContext } from 'vtex.search-page-context/SearchPageContext'
import { generateBlockClass } from '@vtex/css-handles'

import styles from './searchResult.css'

const NotFoundLayout = ({
  children,
  searchQuery,
  maxItemsPerPage,
  map,
  params,
  priceRange,
  orderBy,
  blockClass,
}) => {
  const context = useMemo(
    () => ({
      searchQuery,
      maxItemsPerPage,
      map,
      params,
      priceRange,
      orderBy,
    }),
    [map, maxItemsPerPage, orderBy, params, priceRange, searchQuery]
  )

  return (
    <SearchPageContext.Provider value={context}>
      <div
        className={`${generateBlockClass(
          styles['notFound--layout'],
          blockClass
        )} flex flex-column flex-grow-1`}
      >
        {children}
      </div>
    </SearchPageContext.Provider>
  )
}

export default NotFoundLayout
