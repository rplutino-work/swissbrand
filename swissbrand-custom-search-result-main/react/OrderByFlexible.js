import React from 'react'
import { useSearchPage } from 'vtex.search-page-context/SearchPageContext'

import OrderBy from './OrderBy'

import styles from './searchResult.css'

const withSearchPageContextProps = Component => props => {
  const { orderBy } = useSearchPage()
  if (orderBy == null) {
    return null
  }

  return (
    <div className={styles['orderBy--layout']}>
      <Component {...props} orderBy={orderBy} />
    </div>
  )
}

const OrderByFlexible = withSearchPageContextProps(OrderBy)

OrderByFlexible.schema = {
  title: 'admin/editor.search-result.ordination.sort-by',
}

export default OrderByFlexible
