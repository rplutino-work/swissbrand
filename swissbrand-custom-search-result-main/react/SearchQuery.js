import SearchQuery from './components/SearchQuery'
export default SearchQuery
