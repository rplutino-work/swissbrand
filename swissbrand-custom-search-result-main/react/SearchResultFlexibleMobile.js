import React from 'react'
import SearchResultFlexible from './SearchResultFlexible'

const SearchResultFlexibleMobile = props => {
  return <SearchResultFlexible {...props} />
}

SearchResultFlexibleMobile.schema = {
  title: 'admin/editor.search-result-mobile.title',
}

export default SearchResultFlexibleMobile
