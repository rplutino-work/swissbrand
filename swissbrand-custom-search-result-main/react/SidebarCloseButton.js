import SidebarCloseButton from './components/SidebarCloseButton'
export default SidebarCloseButton
