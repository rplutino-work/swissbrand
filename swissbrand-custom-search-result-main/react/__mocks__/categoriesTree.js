export default [
  {
    name: 'Eletrônicos',
    link: '/Eletronicos',
    quantity: 0,
    children: [
      {
        name: 'Smartphones',
        link: '/Eletronicos/Smartphones',
        quantity: 0,
      },
      {
        name: 'Videogames',
        link: '/Eletronicos/Videogames',
        quantity: 0,
      },
      {
        name: 'TVs',
        link: '/Eletronicos/TVs',
        quantity: 0,
      },
    ],
  },
  {
    name: 'Livros',
    link: '/Livros',
    quantity: 0,
    children: [
      {
        name: 'HQs e Mangas',
        link: '/Livros/HQs e Mangas',
        quantity: 0,
      },
    ],
  },
  {
    name: 'Roupas',
    link: '/Roupas',
    quantity: 0,
    children: [
      {
        name: 'Blusas',
        link: '/Roupas/Blusas',
        quantity: 0,
      },
      {
        name: 'Calças',
        link: '/Roupas/Calcas',
        quantity: 0,
      },
      {
        name: 'Moletom',
        link: '/Roupas/Moletom',
        quantity: 0,
      },
    ],
  },
]

export const numberOfFilters = 10
