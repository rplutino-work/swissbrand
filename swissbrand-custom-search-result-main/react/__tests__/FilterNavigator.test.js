/* eslint-env jest */
import React from 'react'
import { render } from '@vtex/test-tools/react'

import categoriesTree from 'categoriesTree'

import FilterNavigator from '../FilterNavigator'
import QueryContext from '../components/QueryContext'

import { useRuntime } from '../__mocks__/vtex.render-runtime'
import FilterNavigatorContext from '../components/FilterNavigatorContext'
const mockUseRuntime = useRuntime

const mockNavigate = jest.fn()
const mockSetQuery = jest.fn()
beforeEach(() => {
  jest.clearAllMocks()

  mockUseRuntime.mockImplementation(() => ({
    navigate: mockNavigate,
    setQuery: mockSetQuery,
    getSettings: () => ({}),
  }))
})

describe('<FilterNavigator />', () => {
  const renderComponent = (customProps = { query: 'clothing' }) => {
    const props = {
      map: 'c',
      tree: categoriesTree,
      queryArgs: { query: customProps.query, map: 'c' },
      ...customProps,
    }

    return render(
      <QueryContext.Provider
        value={{ query: customProps.query, map: props.map }}
      >
        <FilterNavigatorContext.Provider
          value={{ query: customProps.query, map: props.map }}
        >
          <FilterNavigator {...props} />
        </FilterNavigatorContext.Provider>
      </QueryContext.Provider>
    )
  }

  it('should match snapshot with all', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should display related categories', () => {
    const { asFragment } = renderComponent({ query: 'Livros' })

    expect(asFragment()).toMatchSnapshot()
  })
})
