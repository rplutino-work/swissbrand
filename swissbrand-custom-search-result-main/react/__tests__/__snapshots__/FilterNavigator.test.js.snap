// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterNavigator /> should display related categories 1`] = `
<DocumentFragment>
  <div
    class=" filtersWrapper"
  >
    <div
      class="filter__container filter__container--title bb b--muted-4"
    >
      <h5
        class="filterMessage t-heading-5 mv5"
      >
        <span>
          Filters
        </span>
      </h5>
    </div>
    <div
      class="filter__container filter__container--c bb b--muted-4"
    >
      <div
        class="filter pt4"
      >
        <div
          class="filterTitle t-mini c-muted-2 flex items-center justify-between"
        >
          <span>
            Departments
          </span>
        </div>
      </div>
      <div
        class="categoriesContainer pb5 flex flex-column"
      >
        <div
          class="categoryGroup mt4"
        >
          <div
            class="categoryParent flex items-center pointer"
            role="button"
            tabindex="-1"
          >
            <div
              class="flex-grow-1 dim"
            >
              <span
                class="categoryItemName f5 c-on-base"
              >
                Eletrônicos
              </span>
            </div>
          </div>
          <div
            class="categoryItemChildrenContainer pl5 pl0-ns"
          >
            <div
              class="mt2"
            >
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Smartphones
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Videogames
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                TVs
              </div>
            </div>
          </div>
        </div>
        <div
          class="categoryGroup mt4"
        >
          <div
            class="categoryParent flex items-center pointer"
            role="button"
            tabindex="-1"
          >
            <div
              class="flex-grow-1 dim"
            >
              <span
                class="categoryItemName f5 c-on-base"
              >
                Livros
              </span>
            </div>
          </div>
          <div
            class="categoryItemChildrenContainer pl5 pl0-ns"
          >
            <div
              class="mt2"
            >
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                HQs e Mangas
              </div>
            </div>
          </div>
        </div>
        <div
          class="categoryGroup mt4"
        >
          <div
            class="categoryParent flex items-center pointer"
            role="button"
            tabindex="-1"
          >
            <div
              class="flex-grow-1 dim"
            >
              <span
                class="categoryItemName f5 c-on-base"
              >
                Roupas
              </span>
            </div>
          </div>
          <div
            class="categoryItemChildrenContainer pl5 pl0-ns"
          >
            <div
              class="mt2"
            >
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Blusas
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Calças
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Moletom
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
     Extension Point: shop-review-summary 
  </div>
</DocumentFragment>
`;

exports[`<FilterNavigator /> should match snapshot with all 1`] = `
<DocumentFragment>
  <div
    class=" filtersWrapper"
  >
    <div
      class="filter__container filter__container--title bb b--muted-4"
    >
      <h5
        class="filterMessage t-heading-5 mv5"
      >
        <span>
          Filters
        </span>
      </h5>
    </div>
    <div
      class="filter__container filter__container--c bb b--muted-4"
    >
      <div
        class="filter pt4"
      >
        <div
          class="filterTitle t-mini c-muted-2 flex items-center justify-between"
        >
          <span>
            Departments
          </span>
        </div>
      </div>
      <div
        class="categoriesContainer pb5 flex flex-column"
      >
        <div
          class="categoryGroup mt4"
        >
          <div
            class="categoryParent flex items-center pointer"
            role="button"
            tabindex="-1"
          >
            <div
              class="flex-grow-1 dim"
            >
              <span
                class="categoryItemName f5 c-on-base"
              >
                Eletrônicos
              </span>
            </div>
          </div>
          <div
            class="categoryItemChildrenContainer pl5 pl0-ns"
          >
            <div
              class="mt2"
            >
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Smartphones
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Videogames
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                TVs
              </div>
            </div>
          </div>
        </div>
        <div
          class="categoryGroup mt4"
        >
          <div
            class="categoryParent flex items-center pointer"
            role="button"
            tabindex="-1"
          >
            <div
              class="flex-grow-1 dim"
            >
              <span
                class="categoryItemName f5 c-on-base"
              >
                Livros
              </span>
            </div>
          </div>
          <div
            class="categoryItemChildrenContainer pl5 pl0-ns"
          >
            <div
              class="mt2"
            >
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                HQs e Mangas
              </div>
            </div>
          </div>
        </div>
        <div
          class="categoryGroup mt4"
        >
          <div
            class="categoryParent flex items-center pointer"
            role="button"
            tabindex="-1"
          >
            <div
              class="flex-grow-1 dim"
            >
              <span
                class="categoryItemName f5 c-on-base"
              >
                Roupas
              </span>
            </div>
          </div>
          <div
            class="categoryItemChildrenContainer pl5 pl0-ns"
          >
            <div
              class="mt2"
            >
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Blusas
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Calças
              </div>
              <div
                class="categoryItemChildren ph5 ph3-ns pv5 pv1-ns lh-copy pointer hover-bg-muted-5 c-muted-1"
                role="link"
                tabindex="0"
              >
                Moletom
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
     Extension Point: shop-review-summary 
  </div>
</DocumentFragment>
`;
